hfile.c

```diff
@@ -615,6 +615,47 @@ static hFILE *hopen_fd(const char *filename, const char *mode)
     return NULL;
 }
 
+static hFILE *hpreload_fd(const char *filename, const char *mode)
+{
+    if(!strchr(mode, 'r'))
+    {
+        return NULL;
+    }
+    
+    hFILE_fd *fp = NULL;
+    FILE *file = fopen(filename, mode);
+    if (!file) goto error;
+
+    fseek(file, 0, SEEK_END);
+    int len = ftell(file);
+    fseek(file, 0, SEEK_SET);
+
+    char* buffer = malloc(len);
+    if(buffer == NULL)
+    {
+        errno = ENOMEM;
```

Comment:
Manually setting errno isn't recommended.  Malloc will do this itself, as would the fread below.  Indeed we're obscuring the fread return as it may have other more meaningful errno values.